### ISSUE TITLE
Harbor resource status always in InProgress and Failed

### ISSUE THREAD

**Issue by heww** [goharbor/harbor-operator#468]
Run `kubectl apply -f manifests/samples/full_stack_fs.yaml` to install a harbor cluster resource.

After the pods are all running, then get the status harbor resource, but it's always in `InProgress` and `Failed`.

```
NAME                                                              READY   STATUS    RESTARTS   AGE
harborcluster-sample-harbor-harbor-core-79ffd9b756-5bd8q          1/1     Running   0          28m
harborcluster-sample-harbor-harbor-jobservice-55cb9448d8-hsm4w    1/1     Running   0          27m
harborcluster-sample-harbor-harbor-portal-5fdcb748dd-8cwtd        1/1     Running   0          28m
harborcluster-sample-harbor-harbor-registry-7b46d884fb-df55f      1/1     Running   0          28m
harborcluster-sample-harbor-harbor-registryctl-668d8cdb68-tnlpz   1/1     Running   0          27m
postgresql-cluster-sample-ns-harborcluster-sample-0               1/1     Running   0          31m
postgresql-cluster-sample-ns-harborcluster-sample-1               1/1     Running   0          29m
rfr-harborcluster-sample-redis-0                                  1/1     Running   0          31m
rfs-harborcluster-sample-redis-844c459c68-6xd8w                   1/1     Running   0          31m
```

```yaml
status:
  conditions:
  - message: An error occurred and may be recovered
    reason: recoverrableError
    status: "True"
    type: InProgress
  - message: Ingress.networking.k8s.io cluster-sample-ns/harborcluster-sample-harbor-harbor
    reason: dependencyStatus
    status: "True"
    type: Failed
  observedGeneration: 1
  operator:
    controllerName: harbor
    controllerVersion: dev
```

**Comment by steven-zou** [comment]
kubectl get
```
~/code/harbor-operator$ k8s get harbor -o wide -n cluster-sample-ns
NAME                          PUBLIC URL                            AGE   FAILURE
harborcluster-sample-harbor   https://harbor.10.192.209.17.xip.io   15m   Ingress.networking.k8s.io cluster-sample-ns/harborcluster-sample-harbor-harbor
```

**Comment by heww** [comment]
Closed by #470
